### PR TITLE
[QA-454] Utility function documentation

### PR DESCRIPTION
**Pull request by Tom-Dann**
### What?
Update documentation for utility functions and add an action to publish the automatically generated TypeDoc documentation to GitHub pages

#### Changes:
- `.github/workflows/typedoc-publish.yml`: Added action to build TypeDoc documentation and publish to Github pages
- `deploy/scripts/package.json`: Added `typedoc` to project dev dependencies
- `deploy/scripts/src/common/utils/README.md`: Added a short README for the utils folder
- `deploy/scripts/src/common/utils/**/*.ts`: Converted comments to JSDoc/TSDoc format where applicable

---

### Why?
To improve developer experience and provide enhanced documentation on custom utility functions and types

---

### Related:
- #532

### CODE REVIEW

File: deploy/scripts/src/cri-lime/test.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import exec from 'k6/execution'
 import { timeRequest } from '../common/utils/request/timing'
 import { isStatusCode200, isStatusCode302, pageContentCheck } from '../common/utils/checks/assertions'
 import { sleepBetween } from '../common/utils/sleep/sleepBetween'
+import { getEnv } from '../common/utils/config/environment-variables'
```
Comment (abhinaysrivastavaperf) [inline deploy/scripts/src/cri-lime/test.ts:12]: I have included this change in `[PR-544](https://github.com/govuk-one-login/performance-testing/pull/544)`. We can keep the scope of this PR to utility function documentation and merge this change via the other PR if that's okay.

File: deploy/scripts/src/cri-kiwi/BAV.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import { isStatusCode200, pageContentCheck } from '../common/utils/checks/assert
 import { sleepBetween } from '../common/utils/sleep/sleepBetween'
 import { bankingPayload } from './data/BAVdata'
 import { getAuthorizeauthorizeLocation, getClientID, getCodeFromUrl, getAccessToken } from './utils/authorization'
+import { getEnv } from '../common/utils/config/environment-variables'
```
Comment (abhinaysrivastavaperf) [inline deploy/scripts/src/cri-kiwi/BAV.ts:12]: Same as previous comment - I will include this change in `[PR-544](https://github.com/govuk-one-login/performance-testing/pull/544)` (Kiwi scripts still pending). 
We can keep the scope of this PR to utility function documentation and merge this change via the other PR if that's okay.

File: deploy/scripts/src/cri-kiwi/f2f-cic.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ import { timeRequest } from '../common/utils/request/timing'
 import { isStatusCode200, pageContentCheck } from '../common/utils/checks/assertions'
 import { sleepBetween } from '../common/utils/sleep/sleepBetween'
 import { getAuthorizeauthorizeLocation, getClientID, getCodeFromUrl, getAccessToken } from './utils/authorization'
+import { getEnv } from '../common/utils/config/environment-variables'
```
Comment (abhinaysrivastavaperf) [inline deploy/scripts/src/cri-kiwi/f2f-cic.ts:12]: Same as previous comment - I will include this change in `[PR-544](https://github.com/govuk-one-login/performance-testing/pull/544)` (Kiwi scripts still pending). 
We can keep the scope of this PR to utility function documentation and merge this change via the other PR if that's okay.

**Review by abhinaysrivastavaperf — CHANGES_REQUESTED** [review]

File: deploy/scripts/src/cri-lime/test.ts (hunk 1 above)
Comment (Tom-Dann) [reply deploy/scripts/src/cri-lime/test.ts:12]: These changes are actually needed to make sure the TsDoc builds, it shouldn't matter which PR it is merged in. Might resolve in a merge conflict that needs resolution, but it should be simple to resolve

**Review by abhinaysrivastavaperf — APPROVED** [review]

**Pull request merged**